### ISSUE TITLE
icp across zones creates unexpected paths

### ISSUE THREAD

**Issue by keithj** [irods/irods#3547]
iRODS 4.1.0, icommands 4.1.10.

Expected behaviour: `icp -r /foo /bar` to copy the contents of `foo` into `/bar` such that collection `/foo/x/` is copied to `/bar/x/` and file `/foo/y` is copied to `/bar/y` etc.

Observed behaviour: `/foo/x/` is copied to `/bar//foo/x/`, the collection `/bar//foo/x` appears not to be parseable by `irm` in order to delete it.

````
$ ipwd
/Sanger1/home/kdj
$ iput README /seq/test/kdj/README
$
$ icp -r /seq/test/kdj/ .
Level 0: collection '/Sanger1/home/kdj//seq/test/kdj' is unknown
Level 1: collection '/Sanger1/home/kdj//seq/test/kdj' is unknown

$ ils -r
/Sanger1/home/kdj//seq:
  C- /Sanger1/home/kdj//seq/test
/Sanger1/home/kdj//seq/test:
  C- /Sanger1/home/kdj//seq/test/kdj
/Sanger1/home/kdj//seq/test/kdj:
  README

$ irm -rf /Sanger1/home/kdj//seq
ERROR: rmUtil: srcPath /Sanger1/home/kdj/seq does not exist
$ irm -rf /Sanger1/home/kdj/\/seq
ERROR: rmUtil: srcPath /Sanger1/home/kdj/seq does not exist
````

**Comment by trel** [comment]
Related to https://github.com/irods/irods/issues/2010 and https://github.com/irods/irods/issues/3543

Need consistent path canonicalization in a few places.

**Comment by andrew-irods** [comment]
This issue will be addressed after issue #4057 is fixed.

**Comment by alanking** [comment]
This has been resolved along with #4057.
